homeassistant/components/flume/binary_sensor.py

```diff
@@ -0,0 +1,159 @@
+"""Flume binary sensors."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    DOMAIN,
+    FLUME_AUTH,
+    FLUME_DEVICES,
+    FLUME_TYPE_BRIDGE,
+    FLUME_TYPE_SENSOR,
+    KEY_DEVICE_ID,
+    KEY_DEVICE_LOCATION,
+    KEY_DEVICE_LOCATION_NAME,
+    KEY_DEVICE_TYPE,
+    NOTIFICATION_HIGH_FLOW,
+    NOTIFICATION_LEAK_DETECTED,
+)
+from .coordinator import (
+    FlumeDeviceConnectionUpdateCoordinator,
+    FlumeNotificationDataUpdateCoordinator,
+)
+from .entity import FlumeEntity
+
+
+@dataclass
+class FlumeBinarySensorRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    event_rule: str
+
+
+@dataclass
+class FlumeBinarySensorEntityDescription(
+    BinarySensorEntityDescription, FlumeBinarySensorRequiredKeysMixin
+):
+    """Describes a binary sensor entity."""
+
+
+FLUME_BINARY_NOTIFICATION_SENSORS: tuple[FlumeBinarySensorEntityDescription, ...] = (
+    FlumeBinarySensorEntityDescription(
+        key="leak",
+        name="Leak detected",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        event_rule=NOTIFICATION_LEAK_DETECTED,
+        icon="mdi:pipe-leak",
+    ),
+    FlumeBinarySensorEntityDescription(
+        key="flow",
+        name="High flow",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        event_rule=NOTIFICATION_HIGH_FLOW,
+        icon="mdi:waves",
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up a Flume binary sensor.."""
+
+    flume_domain_data = hass.data[DOMAIN][config_entry.entry_id]
+
+    flume_auth = flume_domain_data[FLUME_AUTH]
+    flume_devices = flume_domain_data[FLUME_DEVICES]
+
+    flume_entity_list: list[
+        FlumeNotificationBinarySensor | FlumeConnectionBinarySensor
+    ] = []
+
+    connection_coordinator = FlumeDeviceConnectionUpdateCoordinator(
+        hass=hass, flume_devices=flume_devices
+    )
+    notification_coordinator = FlumeNotificationDataUpdateCoordinator(
+        hass=hass, auth=flume_auth
+    )
+
+    for device in flume_devices.device_list:
+        device_id = device[KEY_DEVICE_ID]
+        device_location_name = device[KEY_DEVICE_LOCATION][KEY_DEVICE_LOCATION_NAME]
```

Comment:
I think there was another PR where this key could possibly be missing and we need to skip ones without a location.  Maybe move that into a new function which returns only valid devices and use it everywhere?

Reply:
Was it this PR?
https://github.com/home-assistant/core/pull/79456

Reply:
I just did a little list comprehension instead of dropping a new method in:

```
    flume_devices = [
        device
        for device in flume_domain_data[FLUME_DEVICES].device_list
        if KEY_DEVICE_LOCATION_NAME in device[KEY_DEVICE_LOCATION]
        and device[KEY_DEVICE_TYPE] == FLUME_TYPE_SENSOR
    ]
```